### ISSUE TITLE
HelmChart 0.4.0 problems with prometheus config (metric_relabel_configs)

### ISSUE THREAD

**Issue by tvalchev2** [newrelic-experimental/monitoring-kubernetes-with-opentelemetry#91]
[NOTE]: # " ^^ Provide a general summary of the issue in the title above. ^^ "

## Description
We are using the 0.4.0 Release of nrotelk8s which added the configurable filelog and important metrics config to the values.yaml. However the DaemonSets and Deployments are not starting because of faulty prometheus config.

## Steps to Reproduce
Install the 0.4.0 nrotelk8s Release with important metrics definitions. We just uncommented the default important metrics that were already defined in the values.yaml. Containers of DS and Deployments are not starting due to internal errors with the prometheus config.

## Expected Behavior
Everything should work as intended and the metric configuration should be applied to the collectors.

## Relevant Logs / Console output
Here is an example of the error Logs from the DaemonSet (error is the same for Deployment Rec and Smp):
```
Error: failed to get config: cannot unmarshal the configuration: 1 error(s) decoding:

* error decoding 'receivers': error reading configuration for "prometheus/self": prometheus receiver failed to unmarshal yaml to prometheus config: yaml: unmarshal errors:
  line 5: field metric_relabel_configs not found in type struct { Targets []string "yaml:\"targets\""; Labels model.LabelSet "yaml:\"labels\"" }
2023/11/28 09:09:16 collector server run finished with error: failed to get config: cannot unmarshal the configuration: 1 error(s) decoding:

* error decoding 'receivers': error reading configuration for "prometheus/self": prometheus receiver failed to unmarshal yaml to prometheus config: yaml: unmarshal errors:
  line 5: field metric_relabel_configs not found in type struct { Targets []string "yaml:\"targets\""; Labels model.LabelSet "yaml:\"labels\"" }
Error: failed to get config: cannot unmarshal the configuration: 1 error(s) decoding:

* error decoding 'receivers': error reading configuration for "prometheus/self": prometheus receiver failed to unmarshal yaml to prometheus config: yaml: unmarshal errors:
  line 5: field metric_relabel_configs not found in type struct { Targets []string "yaml:\"targets\""; Labels model.LabelSet "yaml:\"labels\"" }
2023/11/28 09:14:29 collector server run finished with error: failed to get config: cannot unmarshal the configuration: 1 error(s) decoding:

* error decoding 'receivers': error reading configuration for "prometheus/self": prometheus receiver failed to unmarshal yaml to prometheus config: yaml: unmarshal errors:
  line 5: field metric_relabel_configs not found in type struct { Targets []string "yaml:\"targets\""; Labels model.LabelSet "yaml:\"labels\"" }
```
## Your Environment

- Versions of your OTel collectors: 0.4.0
- Kubernetes manifest files (only the relevant configuration for important metrics):
```
deployment: 
  prometheus:
    receiver:
      self:
        # Scrape interval
        scrapeInterval: 60s
        # Keeps only the most important metrics and drops the rest of the scraped metrics
        importantMetrics:
          - otelcol_exporter_queue_size
          - otelcol_exporter_queue_capacity
          - otelcol_processor_dropped_metric_points
          - otelcol_processor_dropped_spans
          - otelcol_processor_dropped_log_records
          - otelcol_exporter_enqueue_failed_metric_points
          - otelcol_exporter_enqueue_failed_spans
          - otelcol_exporter_enqueue_failed_log_records
          - otelcol_receiver_refused_metric_points
          - otelcol_receiver_refused_spans
          - otelcol_receiver_refused_log_records
          - otelcol_exporter_refused_metric_points
          - otelcol_exporter_refused_spans
          - otelcol_exporter_refused_log_records
    # Sampler collectors
    sampler:
      # Collector scraping its own metrics
      self:
        # Scrape interval
        scrapeInterval: 60s
        # Keeps only the most important metrics and drops the rest of the scraped metrics
        importantMetrics:
          - otelcol_exporter_queue_size
          - otelcol_exporter_queue_capacity
          - otelcol_processor_dropped_metric_points
          - otelcol_processor_dropped_spans
          - otelcol_processor_dropped_log_records
          - otelcol_exporter_enqueue_failed_metric_points
          - otelcol_exporter_enqueue_failed_spans
          - otelcol_exporter_enqueue_failed_log_records
          - otelcol_receiver_refused_metric_points
          - otelcol_receiver_refused_spans
          - otelcol_receiver_refused_log_records
          - otelcol_exporter_refused_metric_points
          - otelcol_exporter_refused_spans
          - otelcol_exporter_refused_log_records
daemonset:
  # Specific Prometheus configuration
  prometheus:
    # Collector scraping its own metrics
    self:
      # Scrape interval
      scrapeInterval: 60s
      # Keeps only the most important metrics and drops the rest of the scraped metrics
      importantMetrics:
        - otelcol_exporter_queue_size
        - otelcol_exporter_queue_capacity
        - otelcol_processor_dropped_metric_points
        - otelcol_processor_dropped_spans
        - otelcol_processor_dropped_log_records
        - otelcol_exporter_enqueue_failed_metric_points
        - otelcol_exporter_enqueue_failed_spans
        - otelcol_exporter_enqueue_failed_log_records
        - otelcol_receiver_refused_metric_points
        - otelcol_receiver_refused_spans
        - otelcol_receiver_refused_log_records
        - otelcol_exporter_refused_metric_points
        - otelcol_exporter_refused_spans
        - otelcol_exporter_refused_log_records
statefulset: 
  # Specific Prometheus configuration
  prometheus:
    # Collector scraping its own metrics
    self:
      # Scrape interval
      scrapeInterval: 60s
      # Keeps only the most important metrics and drops the rest of the scraped metrics
      importantMetrics:
        - otelcol_exporter_queue_size
        - otelcol_exporter_queue_capacity
        - otelcol_processor_dropped_metric_points
        - otelcol_processor_dropped_spans
        - otelcol_processor_dropped_log_records
        - otelcol_exporter_enqueue_failed_metric_points
        - otelcol_exporter_enqueue_failed_spans
        - otelcol_exporter_enqueue_failed_log_records
        - otelcol_receiver_refused_metric_points
        - otelcol_receiver_refused_spans
        - otelcol_receiver_refused_log_records
        - otelcol_exporter_refused_metric_points
        - otelcol_exporter_refused_spans
        - otelcol_exporter_refused_log_records

    # Kube API server specific configuration
    kubeApiServer:
      # Scrape interval
      scrapeInterval: 60s
      # Keeps only the most important metrics and drops the rest of the scraped metrics
      importantMetrics:
        - apiserver_request_duration_seconds_.*
        - apiserver_request_total
        - workqueue_adds_total
        - workqueue_depth
        - apiserver_current_inflight_requests
        - apiserver_dropped_requests_total

    # Nodes specific configuration
    nodes:
      # Scrape interval
      scrapeInterval: 60s

    # cAdvisor specific configuration
    cAdvisor:
      # Scrape interval
      scrapeInterval: 60s
      # Keeps only the most important metrics and drops the rest of the scraped metrics
      importantMetrics:
        - container_cpu_usage_seconds_total
        - container_memory_usage_bytes
        - container_fs_reads_total
        - container_fs_writes_total
        - container_network_receive_bytes_total
        - container_network_transmit_bytes_total

    # Core DNS specific configuration
    coreDns:
      # Scrape interval
      scrapeInterval: 60s
      # Keeps only the most important metrics and drops the rest of the scraped metrics
      importantMetrics:
        - coredns_dns_request_duration_seconds
        - coredns_dns_requests_total
        - coredns_dns_responses_total
        - coredns_panics_total
        - coredns_cache_hits_total

    # Node exporter specific configuration
    nodeExporter:
      # Scrape interval
      scrapeInterval: 60s
      # Keeps only the most important metrics and drops the rest of the scraped metrics
      importantMetrics:
        - node_cpu_seconds_total
        - node_memory_MemTotal_bytes
        - node_memory_MemFree_bytes
        - node_memory_Cached_bytes
        - node_memory_Buffers_bytes
        - node_filesystem_avail_bytes
        - node_filesystem_size_bytes

    # Kube state metrics specific configuration
    kubeStateMetrics:
      # Scrape interval
      scrapeInterval: 60s
      # Keeps only the most important metrics and drops the rest of the scraped metrics
      importantMetrics:
        - kube_pod_info
        - kube_pod_status_phase
        - kube_pod_container_status_waiting
        - kube_pod_status_scheduled_time
        - kube_daemonset_created
        - kube_daemonset_status_number_available
        - kube_deployment_status_replicas_unavailable
        - kube_deployment_status_condition
        - kube_replicaset_status_ready_replicas
        - kube_replicaset_status_replicas
        - kube_statefulset_replicas
        - kube_statefulset_status_replicas
        - kube_pod_start_time
        - kube_pod_status_reason
        - kube_pod_container_resource_limits
        - kube_pod_status_scheduled
        - kube_pod_container_resource_requests
        - kube_pod_status_container_ready_time
        - kube_pod_container_status_terminated
        - kube_pod_container_status_ready
        - kube_statefulset_status_replicas_updated
        - kube_statefulset_status_replicas_ready
        - kube_daemonset_status_desired_number_scheduled
        - kube_deployment_status_replicas_ready
        - kube_pod_container_state_started
        - kube_statefulset_created
        - kube_daemonset_status_updated_number_scheduled
        - kube_statefulset_status_replicas_current
        - kube_daemonset_status_number_ready
        - kube_pod_container_status_running
        - kube_pod_created
        - kube_pod_container_status_restarts_total
        - kube_daemonset_status_number_unavailable
```
## Additional information
I compared the results from 0.4.0 with the working 0.3.2 Release and noticed, that the metric_release_configs is potentially indented wrong, it must be on the same level as static_configs, directly under scrape_configs for example:
![image](https://github.com/newrelic-experimental/monitoring-kubernetes-with-opentelemetry/assets/101270326/374a86d4-8136-4682-938b-556fd75d7a07)

**Comment by utr1903** [comment]
@tvalchev2 Thanks a lot for the very detailed explanation!

The bug is caused by the following typo: `{{- end -}}` should have been `{{- end }}` at the end of the `importantMetrics` loop... That extra `-` is the reason why the indentation gets messed up.

`0.4.1` is on the way today!

**Comment by utr1903** [comment]
Fixed with #92.
